### ISSUE TITLE
Lint/UnusedMethodArgument and Lint/UselessAssignment triggered incorrectly when binding is called

### ISSUE THREAD

**Issue by oshi-shinobu** [bbatsov/rubocop#1713]
Sample code:

```ruby
def t(b)
  p b.local_variable_get(:v)
  p b.local_variable_get(:a)
end

def m(v)
  a = 'NOOP'
  t binding
end

m 0

# Prints:
# 0
# "NOOP"
```

When `binding` is used, all variables in that scope should be considered in use, regardless if other methods use them or not.

Allowing Rubocop to "correct" the names of the unused arguments means breaking other methods, especially when dealing with templates (such as ERB's).

**Comment by bquorning** [comment]
Related to #1637.

**Comment by bbatsov** [comment]
Indeed. As discussed there - we should discard such offenses if we run into `binding`. PRs welcome! :-)
